### ISSUE TITLE
[Warning] stft with return_complex=False is deprecated.

### ISSUE THREAD

**Issue by Alienpups** [coqui-ai/TTS#2639]
### Describe the bug

If I start to train my model I always get this warning message. Is it just me or is this a thing?

 > TRAINING (2023-05-29 00:46:49)
C:\TTS-Training\lib\site-packages\torch\functional.py:641: UserWarning: stft with return_complex=False is deprecated. In a future pytorch release, stft will return complex tensors for all inputs, and return_complex=False will raise an error.
Note: you can still call torch.view_as_real on the complex output to recover the old return format. (Triggered internally at ..\aten\src\ATen\native\SpectralOps.cpp:867.)
  return _VF.stft(input, n_fft, hop_length, win_length, window,  # type: ignore[attr-defined]

### To Reproduce

I run this command: C:\TTS-Training>python train_vits.py --continue_path "vits_louise-voice-April-21-2023_05+38PM-0000000".
Training starts but I always get warned about some sft error.

 > TRAINING (2023-05-29 00:46:49)
C:\TTS-Training\lib\site-packages\torch\functional.py:641: UserWarning: stft with return_complex=False is deprecated. In a future pytorch release, stft will return complex tensors for all inputs, and return_complex=False will raise an error.
Note: you can still call torch.view_as_real on the complex output to recover the old return format. (Triggered internally at ..\aten\src\ATen\native\SpectralOps.cpp:867.)
  return _VF.stft(input, n_fft, hop_length, win_length, window,  # type: ignore[attr-defined]

### Environment

```shell
{
    "CUDA": {
        "GPU": [],
        "available": false,
        "version": null
    },
    "Packages": {
        "PyTorch_debug": false,
        "PyTorch_version": "2.0.0+cpu",
        "TTS": "0.14.0",
        "numpy": "1.23.5"
    },
    "System": {
        "OS": "Windows",
        "architecture": [
            "64bit",
            "WindowsPE"
        ],
        "processor": "Intel64 Family 6 Model 151 Stepping 2, GenuineIntel",
        "python": "3.10.11",
        "version": "10.0.22631"
    }
}
```

**Comment by Edresson** [comment]
Hi @Alienpups, 

Yeah, it is a thing.

We need to  change [our STFT ](https://github.com/coqui-ai/TTS/blob/d65819422beabd78b124fda91ea1206ec349b06c/TTS/utils/audio/torch_transforms.py#L6) class to use `return_complex=True` and replace:

```
M = o[:, :, :, 0]
P = o[:, :, :, 1]
S = torch.sqrt(torch.clamp(M**2 + P**2, min=1e-8))
```
by

```
o = torch.view_as_real(o)
S = torch.norm(o, p=2, dim=-1)
```
A PR would be welcome :).

**Comment by Alienpups** [comment]
To fix this warning, do I just need to make the changes you mentioned in torch_transforms.py? Did I understand that correctly? And what exactly does PR mean? :)

**Comment by Edresson** [comment]
> To fix this warning, do I just need to make the changes you mentioned in torch_transforms.py? Did I understand that correctly? And what exactly does PR mean? :)
> 
> So I replace these three lines... M = o[:, :, :, 0] P = o[:, :, :, 1] S = torch.sqrt(torch.clamp(M**2 + P**2, min=1e-8))
> 
> with the following? ... o = torch.view_as_real(o) S = torch.norm(o, p=2, dim=-1)

Yes, you also need to use "return_complex=True" on the torch.stft function call.

Sorry, PR means pull request. If you like you can do these changes and then send a pull request and turn a :frog:  TTS contributor :).

**Comment by pivolan** [comment]
I have the same problem, and fix by change edit file python3.10/site-packages/torch/functional.py
at line number 641, just insert this before return statement:
you should find this string:
```
return _VF.stft(input, n_fft, hop_length, win_length, window,  # type: ignore[attr-defined]
```
and insert this before:
```

    if not return_complex:
        return torch.view_as_real(_VF.stft(input, n_fft, hop_length, win_length, window, normalized, onesided, return_complex=True))
```

![image](https://github.com/coqui-ai/TTS/assets/903550/047abf50-c274-44ac-9aac-c6caaf4b0135)
